### PR TITLE
Added GeometryConvertible.simplify(tolerance:)

### PR DESCRIPTION
**Pull request by RCCoop**
Simply adding support for the geos.simplify function :-)

### CODE REVIEW

**Comment by macdrevx** [comment]
Thanks for the contribution, @RCCoop. Could we change the method signature to:

```
func simplify(withTolerance tolerance: Double) throws -> Geometry
```

Otherwise, it looks good to me!

File: GEOSwift/GEOS/GeometryConvertible+GEOS.swift [hunk 1]
```diff
@@ -366,6 +366,17 @@ public extension GeometryConvertible {
         }
         return try Geometry(geosObject: GEOSObject(context: context, pointer: resultPointer))
     }
+
+    // MARK: - Simplify Functions
+
+    func simplify(tolerance: Double) throws -> Geometry {
```
Comment (macdrevx) [inline GEOSwift/GEOS/GeometryConvertible+GEOS.swift:372]:
```suggestion
    func simplify(withTolerance tolerance: Double) throws -> Geometry {
```

File: GEOSwiftTests/GEOS/GeometryConvertible+GEOSTests.swift [hunk 2]
```diff
@@ -906,4 +906,16 @@ final class GeometryConvertible_GEOSTests: XCTestCase {
             }
         }
     }
+
+    // MARK: - Simplify Functions
+
+    func testSimplifyAllTypes() {
+        for geometry in geometryConvertibles {
+            do {
+                _ = try geometry.simplify(tolerance: 0.01)
```
Comment (macdrevx) [inline GEOSwiftTests/GEOS/GeometryConvertible+GEOSTests.swift:915]:
```suggestion
                _ = try geometry.simplify(withTolerance: 0.01)
```

File: GEOSwiftTests/GEOS/GeometryConvertible+GEOSTests.swift [hunk 3]
```diff
@@ -906,4 +906,16 @@ final class GeometryConvertible_GEOSTests: XCTestCase {
...
+            } catch {
+                XCTFail("Unexpected error for \(geometry) simplify(tolerance: 0.01) \(error)")
```
Comment (macdrevx) [inline GEOSwiftTests/GEOS/GeometryConvertible+GEOSTests.swift:917]:
```suggestion
                XCTFail("Unexpected error for \(geometry) simplify(withTolerance: 0.01) \(error)")
```

**Comment by RCCoop** [comment]
Thanks @macdrevx ! I'll make that change and then send it your way. Hoping to do a few other little pull requests in the coming months as I play around with this library. Love it!

**Review by macdrevx — APPROVED** [review]

**Comment by macdrevx** [comment]
Just released in 8.1.0 🎉

**Pull request merged**